pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go

```diff
@@ -86,7 +96,35 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, imc *v1.InMemoryChannel)
 		}
 	}
 	return nil
+}
 
+func (r *Reconciler) patchSubscriberStatus(ctx context.Context, imc *v1.InMemoryChannel) error {
+	after := imc.DeepCopy()
+
+	after.Status.Subscribers = make([]eventingduckv1.SubscriberStatus, 0)
+	for _, sub := range imc.Spec.Subscribers {
+		after.Status.Subscribers = append(after.Status.Subscribers, eventingduckv1.SubscriberStatus{
+			UID:                sub.UID,
+			ObservedGeneration: sub.Generation,
+			Ready:              corev1.ConditionTrue,
+		})
+	}
+	patch, err := duck.CreateMergePatch(imc, after)
```

Comment:
Will the merge patch correctly cause a deletion of subscribers? (I think it depends on how the merge behaviour is configured in the CRD?)

Reply:
Of course I need to write tests for it and so forth, but yes it should because when we create the patch, we start from scratch and only add the current subscribers. So, we spin through the spec.subscribers, and if one is removed it won't get added to this list and hence it will be dropped. But "should" just work.

Reply:
I'm asking because I've seen weird behaviors with slices when the `patchMergeKey` isn't set. Old entries can remain in the existing list, even when removed from the "desired" list. That's how you end up with 2 containers in a Deployment after `apply`-ing a manifest where `containers[0].name` has changed.

ref. https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#notes-on-the-strategic-merge-patch

Reply:
Ah, yeah, we're not doing a strategic merge patch, just a merge patch, but I'll keep this in mind. I need to see if the strategic merge does something different, great point 👍

Reply:
lol @antoineco you were so right :) And I'm laughing because I now remember in the early days of duck typing we had to switch to json patch just for this reason :) I actually had to upstream a patch to get jsonpatch fakes working lol. geez... Switching to JSON patch :)